### DISCUSSION TITLE
Timeout Exception during FTP 4.2.4

### DISCUSSION THREAD

**Discussion by NDSU-UGPTI-ATAC** [udotdevelopment/ATSPM#116] [Signal Controllers]
We are in the process of installing v4.2.4 and when running ftpfromasc3-ftpfromallcontrollers for Econolite controllers it keeps erring out with a TimeoutException. We are able to ping the controllers. What we have tried for troubleshooting is changing port number from 161 to 22 (for sftp) and increasing timeout duration. Anyone else run into the same issue? 

The same program worked for the most part in our v4.2.3 except for some controllers where it couldn't delete .datz files and in some cases where it couldn't download - issues we are expecting 4.2.4 would resolve for us. 

![MicrosoftTeams-image (1)](https://user-images.githubusercontent.com/68076125/108735972-c3c24e00-74f6-11eb-8156-2f969015761d.png)

**Comment by markKHA** [comment]
We have noticed the same timeout issue. It seems like the system is able to download and delete the datZ files, but hits the timeout exceptions in TurnOffAsc3LoggingOverSnmp and TurnOnAsc3LoggingOverSnmp.

It seems like these two functions run regardless of controller type, even though the function calls out ASC3. My understanding is that without toggling the snmp flag off and on to force the controller to generate a new dat/datZ file, we're going to start missing some data. Is that specific to ASC3 controllers, or Cobalt and EOS as well?

**Comment by NDSU-UGPTI-ATAC** [comment]
It appears to be non controller specific as we are unable to download data from any of them. We are able to ping them, login, download, and delete .datZ data via other programs (e.g. WinSCP)

**Comment by ZachParkerCR** [comment]
> It appears to be non controller specific as we are unable to download data from any of them. We are able to ping them, login, download, and delete .datZ data via other programs (e.g. WinSCP)

@NDSU-UGPTI-ATAC , did you find a solution to this issue?  I'm running into a similar problem, while still on version ATSPM 4.2, with Cobalt controllers running on firmware 32.67.30.  The ftpfromallcontrollers is not able to connect to these controllers using normal FTP on port 21 and I cannot find a way to adjust ATSPM to connect on SFTP port 22.  I'm nervous to go through the 4.24 upgrade process if I'm going to run into the same issue on these controllers.

**Comment by NDSU-UGPTI-ATAC** [comment]
> > It appears to be non controller specific as we are unable to download data from any of them. We are able to ping them, login, download, and delete .datZ data via other programs (e.g. WinSCP)
> 
> @NDSU-UGPTI-ATAC , did you find a solution to this issue? I'm running into a similar problem, while still on version ATSPM 4.2, with Cobalt controllers running on firmware 32.67.30. The ftpfromallcontrollers is not able to connect to these controllers using normal FTP on port 21 and I cannot find a way to adjust ATSPM to connect on SFTP port 22. I'm nervous to go through the 4.24 upgrade process if I'm going to run into the same issue on these controllers.

@ZachParkerCR For the controllers that ftpfromallcontrollers does not connect to, I created a WinSCP batch file that runs on the same schedule. Between the ftpfromallcontrollers and my WinSCP script I am able to download and delete .datZ files from all controllers. The only minor inconvenience is that the controllers on my WinSCP script land up in the 'The following signals have had FTP problems' block of the Watchdog email. We tried 4.2.4 and ran into the same issues. Now eagerly waiting for the big 5.0!
Sharma

KSHITIJ SHARMA
Associate Research Fellow -  Traffic Operations  /  Upper Great Plains Transportation Institute

Dept 2880, PO Box 6050  /  Fargo ND 58108-6050
p: 701.231.1086
www.ndsu.edu

**Comment by ZachParkerCR** [comment]
@NDSU-UGPTI-ATAC , out of curiosity - what is the manufacturer and firmware version of the controllers that you cannot connect to with the ATSPM system?  I'm unable to connect to Econolite Cobalt controllers on firmware version 32.67.30 while running ATSPM 4.2.  

Would you be willing to share your batch file with me so I can review your commands and process?

**Comment by ZachParkerCR** [comment]
@NDSU-UGPTI-ATAC , sorry to bug you, but I'm planning to update our environment to 4.24 and I don't want to run into the same issue after all the time spent performing the upgrade.  If you could let me know the firmware version and manufacturer of the controllers you are still having issues with, that will help me compare it to my environment and explain the situation to my traffic management so their expectations are aligned.

**Comment by NDSU-UGPTI-ATAC** [comment]
@ZachParkerCR Econolite Cobalt Controllers running EOS.

**Comment by NDSU-UGPTI-ATAC** [comment]
@ZachParkerCR I have sent you sample batch and text files. We will be testing 5.0 version's ftpfromallcontrollers program when released before upgrading to make sure the sftp works on our controllers.

**Comment by ZachParkerCR** [comment]
> @ZachParkerCR I have sent you sample batch and text files. We will be testing 5.0 version's ftpfromallcontrollers program when released before upgrading to make sure the sftp works on our controllers.

@NDSU-UGPTI-ATAC , thanks again for all your offline help!
